src/query/parser/promql/types.go

```diff
@@ -76,13 +75,12 @@ func NewAggregationOperator(expr *promql.AggregateExpr) (parser.Params, error) {
 	}
 
 	if op == aggregation.BottomKType || op == aggregation.TopKType {
-		paramString := expr.Param.String()
-		floatParam, err := strconv.ParseFloat(paramString, 64)
+		val, err := resolveScalarArgument(expr.Param)
```

Comment:
consider calling this for all aggregation types. Otherwise, picking and choosing will be hard :(

Reply:
Only these two agg functions take scalar args, so not necessary to call it for others